src/master-playlist-controller.js

```diff
@@ -1259,6 +1284,6 @@ export class MasterPlaylistController extends videojs.EventTarget {
     const rate = Config.BUFFER_LOW_WATER_LINE_RATE;
     const max = Math.max(initial, Config.MAX_BUFFER_LOW_WATER_LINE);
 
-    return Math.min(initial + currentTime * rate, max);
+    return Math.min(initial + currentTime * rate, max, this.goalBufferLength());
```

Comment:
is there a specific reason for this? I understood the intention of the low water line to always be less than the goalBufferLength

Reply:
Exactly. This ensures `bufferLowWaterLine() <= goalBufferLength` will always be true.

Reply:
Precisely. With fixed values, the low water line could never be less than the target buffer length. But since the target buffer length is now dynamic, we need to make sure that `bufferLowWaterLine() <= goalBufferLength`